test/exec/test_at_coroutine_exit.cpp

```diff
@@ -0,0 +1,239 @@
+}
+
+namespace {
+exec::task<void> invoke_actions_after_stop(int& result, auto stop) {
+  ++result;
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  co_await stop;
+  ++result;
+}
+
+exec::task<void> invoke_actions_after_stop2(int& result, auto stop) {
+  ++result;
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+
+  co_await stop;
+
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  ++result;
+}
+
+exec::task<void> invoke_actions_after_stop3(int& result, auto stop) {
+  ++result;
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  ++result;
+
+  co_await stop;
+}
+
+exec::task<void> invoke_actions_after_stop4(int& result, auto stop) {
+  ++result;
+  co_await stop;
+
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  ++result;
+}
+}
+
+TEST_CASE("at_coroutine_exit invokes two actions in correct order after stop signal", "[task][at_coroutine_exit]")
+{
+  int result = 0;
+  stopped_scheduler scheduler;
+  stdexec::sync_wait(invoke_actions_after_stop(result, stdexec::schedule(scheduler)));
+  REQUIRE(result == 3);
+
+  result = 0;
+  stdexec::sync_wait(invoke_actions_after_stop2(result, stdexec::schedule(scheduler)));
+  REQUIRE(result == 2);
+
+  result = 0;
+  stdexec::sync_wait(invoke_actions_after_stop3(result, stdexec::schedule(scheduler)));
+  REQUIRE(result == 5);
+
+  result = 0;
+  stdexec::sync_wait(invoke_actions_after_stop4(result, stdexec::schedule(scheduler)));
+  REQUIRE(result == 1);
+}
+
+
+#ifdef __linux__
+namespace {
+exec::task<void> exception_in_action() {
+  co_await exec::at_coroutine_exit([]() -> exec::task<void> {
+    throw test_exception("test");
+    co_return;
+  });
+}
+}
+
+TEST_CASE("at_coroutine_exit terminates after exception within action", "[task][at_coroutine_exit]")
+{
+  REQUIRE_TERMINATE([] { stdexec::sync_wait(exception_in_action()); });
+}
+
+namespace {
+exec::task<void> stop_in_action(auto stop) {
+  co_await exec::at_coroutine_exit([&] { return stop; });
+}
+}
+
+TEST_CASE("at_coroutine_exit terminates after stop within action", "[task][at_coroutine_exit]")
+{
+  stopped_scheduler scheduler;
+  auto stop = stdexec::schedule(scheduler);
+  REQUIRE_TERMINATE([](auto stop) { stdexec::sync_wait(stop_in_action(stop)); }, std::move(stop));
+}
+#endif
```

Comment:
Are these all the test cases from libunifex?

Reply:
No, not all. I will add all of them today!

Reply:
Indeed, the following test fails, because the continuation gets called and I know where I am wrong

```cpp
  int result = 0;
  stdexec::sync_wait(with_continuation(result, test_one_cleanup_action_with_stop(result)));
  REQUIRE(result == 2);
```